### PR TITLE
NoBlindness/NoDarkness in NoRender

### LINKED ISSUE #2242: No Blindness[Suggestion]
### Describe the feature

It would be a module that allows you to remove the blindness effect. So when you have it, there is no overlay.

### Before submitting a suggestion

- [X] This feature doesn't already exist in the client.
(I have checked every module and their settings on the **latest dev build**)

- [X] This wasn't already suggested.
(I have checked every suggestion on github)

- [X] This feature is a valid suggestion.
(I have read the FAQs and Guides)

### PR DESCRIPTION
**Pull request by ThebestkillerTBK**
Resolves #2242 #1950 #1705

### CODE REVIEW

**Comment by Wide-Cat** [comment]
correct me if im wrong but the fog setting in norender already removes blindness and darkness?

**Comment by ThebestkillerTBK** [comment]
This PR fixes the black sky issue and sodium issue

File: src/main/java/meteordevelopment/meteorclient/mixin/BackgroundRendererMixin.java [hunk 1]
```diff
@@ -27,4 +29,9 @@ private static void onApplyFog(Camera camera, BackgroundRenderer.FogType fogType
             }
         }
     }
+
+    @Inject(method = {"getFogModifier(Lnet/minecraft/entity/Entity;F)Lnet/minecraft/client/render/BackgroundRenderer$StatusEffectFogModifier;"}, at = @At("HEAD"), cancellable = true)
+    private static void onGetFogModifier(Entity entity, float tickDelta, CallbackInfoReturnable<Object> info) {
+        if (Modules.get().get(NoRender.class).noBlindness()) info.setReturnValue(null);
```
Comment (SIMULATAN) [inline src/main/java/meteordevelopment/meteorclient/mixin/BackgroundRendererMixin.java:35]: Why are there two settings if they both do literally the same?

**Review by SIMULATAN — CHANGES_REQUESTED** [review]
Small weird design choice that imo needs clarification but otherwise LGTM

File: src/main/java/meteordevelopment/meteorclient/mixin/BackgroundRendererMixin.java (hunk 1 above)
Comment (ThebestkillerTBK) [reply src/main/java/meteordevelopment/meteorclient/mixin/BackgroundRendererMixin.java:35]: It only removes blindness fog

Comment (SIMULATAN) [reply src/main/java/meteordevelopment/meteorclient/mixin/BackgroundRendererMixin.java:35]: Oh, accidentally overlooked these other things and commented on the wrong thing lol, sorry for the wrong review
Doing literally anything on phones is just so terrible :>(

**Review by SIMULATAN — APPROVED** [review]

**Comment by seasnail8169** [comment]
can you make all of your mixin method targets into strings rather than arrays

**Comment by ThebestkillerTBK** [comment]
> can you make all of your mixin method targets into strings rather than arrays

fixed

**Pull request merged**